### PR TITLE
modem_lib: Doc updates for deferred trace processing

### PR DESCRIPTION
**Pull request by balaji-nordic**
New API nrf_modem_trace_processed_callback() added to nrf_modem.h.
Doc updates related to the new API.
Minor corrections in existing documentation.

Signed-off-by: Balaji Srinivasan <balaji.srinivasan@nordicsemi.no>

### CODE REVIEW

File: nrf_modem/doc/ug_nrf_modem_porting_os.rst [hunk 1]
```diff
@@ -134,7 +134,8 @@ nrf_modem_os_trace_put
 
 This function puts the trace string to the desired medium, typically UART.
 However, the medium used to forward and store the traces is up to the implementation and must be initialized correctly before using.
-If you are not interested in traces, they can be ignored, and this function can be empty and simply return.
+Once the traces are processed/stored, the :c:func:`nrf_modem_trace_processed_callback` must called.
```
Comment (gregersrygg) [inline nrf_modem/doc/ug_nrf_modem_porting_os.rst:137]: Missing word: must _be_ called.

Perhaps add a sentence about why this is important.

**Review by gregersrygg — CHANGES_REQUESTED** [review]

File: nrf_modem/doc/ug_nrf_modem_porting_os.rst (hunk 1 above)
Comment (balaji-nordic) [reply nrf_modem/doc/ug_nrf_modem_porting_os.rst]: Done

**Review by lemrey — APPROVED** [review]
Thank you for updating the doc, this looks good to me.

**Review by gregersrygg — APPROVED** [review]
Thank you. LGTM 👍

**Pull request merged**